From 55a90f6deefe018f121b61b25f52a387f1155c9e Mon Sep 17 00:00:00 2001
From: Yin Fengwei <fengwei.yin@intel.com>
Date: Fri, 12 Apr 2019 13:23:19 +0800
Subject: [PATCH 1351/1443] ACRN: hvlog: check whether it's ACRN before init
 hvlog

We need to make sure ACRN hvlog is running under ACRN hypervisor.
Otherwise, hypercall will trigger invalid instruction and trigger
kernel crash

Signed-off-by: Yin Fengwei <fengwei.yin@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/acrn/acrn_hvlog.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/acrn/acrn_hvlog.c b/drivers/acrn/acrn_hvlog.c
index da04aeb..87b1cb2 100644
--- a/drivers/acrn/acrn_hvlog.c
+++ b/drivers/acrn/acrn_hvlog.c
@@ -65,6 +65,8 @@
 #include <linux/vhm/vhm_hypercall.h>
 #include <linux/vhm/acrn_hv_defs.h>
 
+#include <asm/hypervisor.h>
+
 #include "sbuf.h"
 
 #define LOG_ENTRY_SIZE		80
@@ -348,6 +350,11 @@ static int __init acrn_hvlog_init(void)
 	struct acrn_hw_info hw_info;
 	uint64_t cur_logbuf, last_logbuf;
 
+	if (x86_hyper_type != X86_HYPER_ACRN) {
+		pr_err("acrn_hvlog: not running under acrn hypervisor!\n");
+		return -EINVAL;
+	}
+
 	if (!hvlog_buf_base || !hvlog_buf_size) {
 		pr_warn("no fixed memory reserve for hvlog.\n");
 		return 0;
-- 
2.7.4

